src/layouts/components/Editor/LinkBubbleMenu.tsx

```diff
@@ -0,0 +1,93 @@
+import {
+  useDisclosure,
+  Modal,
+  ModalOverlay,
+  ModalContent,
+  ModalHeader,
+  ModalFooter,
+  ModalBody,
+  ModalCloseButton,
+  InputGroup,
+  InputLeftAddon,
+} from "@chakra-ui/react"
+import { Button, Input } from "@opengovsg/design-system-react"
+import { BubbleMenu } from "@tiptap/react"
+import { useState } from "react"
+
+import { useEditorContext } from "contexts/EditorContext"
+
+const LinkButton = () => {
+  const { editor } = useEditorContext()
+  const { onClose, onOpen, isOpen } = useDisclosure()
+  const [href, setHref] = useState("")
+
+  return (
+    <>
+      <button
+        // TODO: Replace with a nice looking button
+        style={{
+          border: "1px solid black",
+          borderRadius: "5px",
+          padding: "1px 6px",
+          backgroundColor: "white",
+        }}
+        type="button"
+        onClick={onOpen}
+      >
+        Change link
+      </button>
+      <Modal isOpen={isOpen} onClose={onClose} size="sm">
+        <ModalOverlay />
+        <ModalContent>
+          <ModalHeader>Update link</ModalHeader>
+          <ModalCloseButton />
+          <ModalBody>
+            <InputGroup size="sm">
+              <InputLeftAddon>https://</InputLeftAddon>
+              <Input
+                defaultValue={(
+                  (editor.getAttributes("link").href as string) || ""
+                )
+                  // NOTE: trim the start of `https://` or `http://`
+                  .replace(/^https?:\/\//, "")}
+                onChange={(event) => setHref(event.target.value)}
+              />
+            </InputGroup>
+          </ModalBody>
+
+          <ModalFooter>
+            <Button
+              colorScheme="blue"
+              mr={3}
+              onClick={() => {
+                editor
+                  .chain()
+                  .focus()
+                  // NOTE: Force `https` by default
+                  .setLink({ href: `https://${href}` })
```

Comment:
what about relative paths ah?

Reply:
hmm good catch! for files, the modal itself uses relative links so it's ok. this _would_ be a problem for updating file extensions. perhaps i'll change so that this happens for absolute links, and not relative.

Reply:
uhm so this is changed upstream? this one impt ya cuz agencies use relative path quite frequently